src/Core/Layer/Layer.js

```diff
@@ -1,3 +1,25 @@
+import { EventDispatcher } from 'three';
+
+
+export const defineLayerProperty = function defineLayerProperty(layer, propertyName, defaultValue, onChange) {
+    var property = layer[propertyName] == undefined ? defaultValue : layer[propertyName];
+    Object.defineProperty(layer,
+        propertyName,
+        { get: () => property,
+            set: (newValue) => {
+                if (property !== newValue) {
+                    const event = { type: `${propertyName}-layer-changed`, previous: {}, new: {} };
```

Comment:
`layer` is redundant in event name `${propertyName}-layer-changed` since the user will register on a layer anyway.

`${propertyName}-changed` or `${propertyName}-property-changed` seem better (I prefer the -property- version). WDYT?